CARv2 MultiIndex.md

```diff
@@ -0,0 +1,40 @@
+# CARv2 MultiIndex
+
+![wip](https://img.shields.io/badge/status-wip-orange.svg?style=flat-square)
+
+## Editors
+
+* [Alan Shaw](https://github.com/alanshaw), [DAG House](https://dag.house/)
+
+## Authors
+
+* [Alan Shaw](https://github.com/alanshaw), [DAG House](https://dag.house/)
+
+## Language
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in [RFC 2119](https://datatracker.ietf.org/doc/html/rfc2119).
+
+## Abstract
+
+A multi-index is a CARv2 index format that allows multiple CARv2 indexes to be grouped together in a single index.
+
+Typically an index with this format is _not_ stored _in_ a CAR file but as a side index.
+
+## Format `0x0402`: MultiIndex
+
+An unsigned varint of `0x0402` at the "Index offset" byte position or more typically for this type of index, at the start of the file, indicates the following bytes should be interpreted as the "MultiIndex" format.
+
+The format allows multiple CARv2 indexes to be grouped together in a single index, and takes the following form:
+
+```sh
+| 0x0402 | count (uint32) | car-multihash | carv2-index | car-multihash | carv2-index | ... |
```

Comment:
why a `uint32` here and not a `varint`? wouldn't a unit32 _have_ to be encoded as 32 bits, while a varint would be more compact for smaller numbers?

Reply:
Yes it'll probably be a low number most of the time in our use case so lets save some bytes!